software/contrib/bernoulli_gates.py

```diff
@@ -0,0 +1,180 @@
+            self.right_port = cv5
+            self.function_port = cv6
+        
+    def get_prob(self):
+        if self.id == 0:
+            self.right_possibility = self.knob_control.percent() + ain.read_voltage()/12
+            self.left_possibility = 1 - self.right_possibility
+        else:
+            self.right_possibility = self.knob_control.percent()
+            self.left_possibility = 1 - self.right_possibility
+            
+    def bar_visualization(self):
+        oled.text(self.text + f':{self.right_possibility:.2f}', self.text_pos[0], self.text_pos[1], 1)
+        oled.rect(int(OLED_WIDTH / 2), 
+                  int((OLED_HEIGHT - BAR_WID) / 2) + self.bar_pos_offset, 
+                  int(self.right_possibility * BAR_LEN), 
+                  BAR_WID, 1) # right
+        oled.fill_rect(int(OLED_WIDTH / 2 - self.left_possibility * BAR_LEN), 
+                       int((OLED_HEIGHT - BAR_WID) / 2) + self.bar_pos_offset, 
+                       int(self.left_possibility * BAR_LEN), 
+                       BAR_WID, 1) # left
+                       
+    def probability_sample(self):
+        self.right_possibility_sampled = self.right_possibility
+        self.left_possibility_sampled = self.left_possibility
+    
+    def triggered_maneuver(self):
+        self.coin = random()
+        if self.mode_flg == 0 or self.mode_flg == 1:
+            if self.coin < (self.right_possibility_sampled):
+                self.left_port.off()
+                self.right_port.on()
+                if self.mode_flg == 0:
+                    # Draw right indicator
+                    oled.fill_rect(int(OLED_WIDTH / 2 + self.right_possibility * BAR_LEN) + 2, 
+                                   int((OLED_HEIGHT - BAR_WID) / 2) + self.bar_pos_offset, 
+                                   INDI_WID, 
+                                   BAR_WID, 1)
+            else:
+                self.left_port.on()
+                self.right_port.off()
+                if self.mode_flg == 0:
+                    # Draw left indicator
+                    oled.rect(int(OLED_WIDTH / 2 - self.left_possibility * BAR_LEN) - INDI_WID - 2, 
+                              int((OLED_HEIGHT - BAR_WID) / 2) + self.bar_pos_offset, 
+                              INDI_WID, 
+                              BAR_WID, 1)
+        else:
+            if self.coin < (self.right_possibility_sampled):
+                self.left_port.toggle()
+                self.right_port.value(self.left_port._duty == 0)
+                 
+        if self.id == 1: self.function_port.value((cv1._duty and cv4._duty) != 0)
+
+    def regular_maneuver(self):
+        if self.mode_flg == 0:
+            oled.text('Tr', self.text1_pos, OLED_HEIGHT-8, 1)
+            sleep_ms(TRG_T)
+            self.left_port.off()
+            self.right_port.off()
+            self.function_port.off()
+        elif self.mode_flg == 1:
+            oled.text('G', self.text1_pos, OLED_HEIGHT-8, 1)
```

Comment:
Minor, but important to fix before we merge; the README refers to this as labeled "L", but the oled display shows "G" when in gate/latch mode. Either is label is fine, but make sure they're consistent.

Reply:
Changed the README to keep the consistency.